### PR TITLE
Fix for eclipse-score#42

### PR DESCRIPTION
**Pull request by Abhishek2581**
Fix for eclipse-score#42 feat(lola): Introduce applicationID for robust crash recovery

This pull request introduces a new, more robust mechanism for application identification in the LoLa (shared memory) binding, which is critical for the crash recovery and partial restart features.

This change introduces an optional applicationID property within the global section of the json configuration file.

- Explicit Identification: Applications can now be assigned a stable, unique, and configuration-defined ID. This is the new primary mechanism for identifying an application instance for crash recovery.

- Fallback for Compatibility: If the applicationID is not provided, the system gracefully falls back to using the process uid. An informational log message is now printed to alert the user that the fallback is active and that unique system UIDs are required.

- Code Simplification: The UidPidMapping class and its related files have been renamed to ApplicationIdPidMapping. The internal logic has been simplified, removing the need for platform-specific preprocessor directives for QNX, as the new applicationID is handled as a standard unsigned integer across all platforms.

**Key Changes**

- Configuration: Added support for global.applicationID in config_parser.cpp.

- Schema: Updated ara_com_config_schema.json to include the new applicationID property with minimum and maximum constraints for robust validation.

- Runtime: The lola::Runtime now determines the application identifier by prioritizing the configured applicationID before falling back to the process uid.

**Testing:**
- Added runtime_application_id_test.cpp to verify the new identifier logic, including boundary-value tests.

- Updated transaction_log_rollback_executor_test.cpp to use the new ApplicationIdPidMapping. This ensures that the proxy-side crash recovery logic, which is responsible for rolling back state and notifying the skeleton of a restart, continues to function correctly with the new identification mechanism.

### CODE REVIEW

File: score/mw/com/impl/bindings/lola/i_runtime.h [hunk 1]
```diff
@@ -45,6 +45,8 @@ class IRuntime : public impl::IRuntimeBinding
     virtual pid_t GetPid() const noexcept = 0;
     /// \brief We need our UID in several locations/frequently. So the runtime shall provide/cache it.
     virtual uid_t GetUid() const noexcept = 0;
+    /// \brief We need our Application ID in several locations/frequently. So the runtime shall provide/cache it.
+    virtual uid_t GetApplicationId() const noexcept = 0;
```
Comment (crimson11) [inline score/mw/com/impl/bindings/lola/i_runtime.h:49]: Returns `std::uint32_t`

File: score/mw/com/impl/bindings/lola/application_id_pid_mapping.h [hunk 2]
```diff
@@ -30,75 +30,76 @@ namespace score::mw::com::impl::lola
...
 // translation units shall be declared in one and only one file.".
 // coverity[autosar_cpp14_m3_2_3_violation] This is false positive. Function is declared only once.
 template <template <class> class AtomicIndirectorType = memory::shared::AtomicIndirectorReal>
 // coverity[autosar_cpp14_m3_2_3_violation] This is false positive. Function is declared only once.
-std::optional<pid_t> RegisterPid(score::containers::DynamicArray<UidPidMappingEntry>::iterator entries_begin,
-                                 score::containers::DynamicArray<UidPidMappingEntry>::iterator entries_end,
-                                 const uid_t uid,
+std::optional<pid_t> RegisterPid(score::containers::DynamicArray<ApplicationIdPidMappingEntry>::iterator entries_begin,
+                                 score::containers::DynamicArray<ApplicationIdPidMappingEntry>::iterator entries_end,
+                                 const uid_t application_id,
```
Comment (crimson11) [inline score/mw/com/impl/bindings/lola/application_id_pid_mapping.h:45]: application_id should be `std::uint32_t`

File: score/mw/com/impl/bindings/lola/application_id_pid_mapping_entry.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+#include "score/mw/com/impl/bindings/lola/application_id_pid_mapping_entry.h"
+
+#include <cstdint>
+
+namespace score::mw::com::impl::lola
+{
+
+std::pair<ApplicationIdPidMappingEntry::MappingEntryStatus, uid_t> ApplicationIdPidMappingEntry::GetStatusAndUidAtomic() noexcept
```
Comment (crimson11) [inline score/mw/com/impl/bindings/lola/application_id_pid_mapping_entry.cpp:20]: `GetStatusAndUidAtomic` -> `GetStatusAndApplicationIdAtomic`

File: score/mw/com/impl/bindings/lola/application_id_pid_mapping_entry.cpp [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    const auto uid_part = static_cast<std::uint32_t>(status_uid & kMaskUid);
+    // Suppress "AUTOSAR C++14 A7-2-1" rule: "An expression with enum underlying type shall only have values
+    // corresponding to the enumerators of the enumeration.".
+    // Callers of this function process only the MappingEntryStatus enumeration values ​​of interest, the rest are
+    // discarded.
+    // coverity[autosar_cpp14_a7_2_1_violation]
+    return {static_cast<ApplicationIdPidMappingEntry::MappingEntryStatus>(status_part), static_cast<uid_t>(uid_part)};
+}
+
+void ApplicationIdPidMappingEntry::SetStatusAndUidAtomic(MappingEntryStatus status, uid_t application_id) noexcept
```
Comment (crimson11) [inline score/mw/com/impl/bindings/lola/application_id_pid_mapping_entry.cpp:34]: `SetStatusAndUidAtomic` -> `SetStatusAndApplicationIdAtomic`

Comment (crimson11) [inline score/mw/com/impl/bindings/lola/application_id_pid_mapping_entry.cpp:34]: application_id should be now `std::uint32_t`

File: score/mw/com/impl/bindings/lola/application_id_pid_mapping_entry.cpp [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+{
+    key_application_id_status_.store(CreateKey(status, application_id));
+}
+
+ApplicationIdPidMappingEntry::key_type ApplicationIdPidMappingEntry::CreateKey(MappingEntryStatus status, uid_t application_id) noexcept
+{
+    ApplicationIdPidMappingEntry::key_type result = static_cast<std::uint16_t>(status);
+    result = result << 32U;
+    static_assert(sizeof(uid_t) <= 4, "For more than 32 bits we cannot guarantee the key to be unique");
```
Comment (crimson11) [inline score/mw/com/impl/bindings/lola/application_id_pid_mapping_entry.cpp:43]: This check/assert should be now moved to the location, where you handle the fallback case (`Runtime::DetermineApplicationIdentifier()`)

**Comment by crimson11** [comment]
BTW: I'm on vacation now for the next 2 weeks!
I'm basically fine with all the changes you did (incorporating my review remarks)!

**Review by lurtz — COMMENTED** [review]
just take my two comments

File: score/mw/com/impl/configuration/global_configuration_mock.h [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+#ifndef SCORE_MW_COM_IMPL_CONFIGURATION_GLOBAL_CONFIGURATION_MOCK_H
+#define SCORE_MW_COM_IMPL_CONFIGURATION_GLOBAL_CONFIGURATION_MOCK_H
+
+#include "score/mw/com/impl/configuration/global_configuration.h"
+#include <gmock/gmock.h>
+
+namespace score::mw::com::impl::configuration
+{
+
+class GlobalConfigurationMock : public GlobalConfiguration
```
[the remark this answers is not in the document]
Comment (Abhishek2581) [reply score/mw/com/impl/configuration/global_configuration_mock.h]: Fixed — removed the unused file as suggested

**Review by crimson11 — APPROVED** [review]
